test/dash.js

```diff
@@ -4,8 +4,11 @@ var parse = require('../');
 var test = require('tape');
 
 test('-', function (t) {
-	t.plan(5);
+	t.plan(8);
 	t.deepEqual(parse(['-n', '-']), { n: '-', _: [] });
+	t.deepEqual(parse(['--nnn', '-']), { nnn: '-', _: [] });
+	t.deepEqual(parse(['-n', '---']), { n: '---', _: [] });
+	t.deepEqual(parse(['--nnn', '---']), { nnn: '---', _: [] });
```

Comment:
i hadn't considered this as the value, but these are good - what about `---nnn=abc`?

Reply:
`---nnn=abc` is not affected by this PR, which only affects processing of a long option with a space separated value which starts with a dash.

For interest, the behaviour before and after is:
```console
% node plain.js ---nnn=abc
{ _: [], '-nnn': 'abc' }
```

Reply:
ok, then `---nnn abc`?